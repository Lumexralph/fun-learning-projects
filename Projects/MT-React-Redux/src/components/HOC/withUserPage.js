import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import HeaderContainer from '../home/HeaderContainer';
import loadUserRequests from '../../actions/loadUserRequests';
import userRequests from '../../actions/userRequests';
import createUserRequest from '../../actions/createUserRequest';
import getSingleRequest from '../../actions/getSingleRequest';
import currentRequest from '../../actions/currentRequest';
import editUserRequest from '../../actions/editUserRequest';

const userPage = (WrappedComponent) => {
  class RequestComponent extends Component {
    state = {
      navLinks: ['Home', 'My Requests', 'Make A Request', 'Logout'],
    }

    componentDidMount() {
      const {
        getUserRequests, storeUserRequests, match, getRequestById, viewedRequest,
        presentRequest,
      } = this.props;
      const { id } = match.params;

      this.setState(previousState => ({
        ...previousState,
        id: presentRequest.request_id,
        request: {
          ...previousState.request,
          title: presentRequest.request_title,
          content: presentRequest.request_content,
          department: presentRequest.department,
        },
      }));

      if (id) {
        getRequestById(id)
          .then((response) => {
            viewedRequest(response.data);
          })
          .catch((error) => {
            const { response } = error;
            this.setState(previousState => ({
              ...previousState,
              message: response.data.message,
            }));
          });
      }

      getUserRequests()
        .then((response) => {
          const requests = response.data.message ? [] : response.data;
          storeUserRequests(requests);
        })
        .catch(error => error);
    }

    handleUserInput = (event) => {
      const { target } = event;
      const { name, value } = target;
      this.setState(previousState => ({
        ...previousState,
        message: null,
        request: {
          ...previousState.request,
          [name]: value,
        },
      }));
    }

    handleFormSubmit = (event) => {
      event.preventDefault();

      const { createRequest, history } = this.props;
      const { request } = this.state;

      createRequest(request)
        .then(() => {
          history.push('/profile');
        })
        .catch((error) => {
          const { response } = error;
          this.setState(previousState => ({
            ...previousState,
            message: response.data.message,
          }));
        });
    }

    handleFormUpdate = (event) => {
      event.preventDefault();

      const { request, id } = this.state;
      const { updateRequest, history } = this.props;

      updateRequest(id, request)
        .then(() => {
          history.push('/profile');
        })
        .catch((error) => {
          const { response } = error;
          this.setState(previousState => ({
            ...previousState,
            message: response.data.message,
          }));
        });
    }

    render() {
      let requestData;
      let handleSubmit;
      const { navLinks, message = null } = this.state;
      const { requests, presentRequest, history, history: { location: { pathname } } } = this.props;

      // set form submit based on create or update request
      if (pathname === '/request/create') {
        handleSubmit = this.handleFormSubmit;
        requestData = null;
      } else {
        handleSubmit = this.handleFormUpdate;
        requestData = presentRequest;
      }

      return (
        <div className="container">
          <HeaderContainer navText={navLinks} history={history} />
          <WrappedComponent
            requests={requests}
            onChange={this.handleUserInput}
            onSubmit={handleSubmit}
            onLoad={this.handleFormDefaultData}
            message={message}
            request={requestData}
            pathname={pathname}
          />
        </div>
      );
    }
  }

  RequestComponent.propTypes = {
    getUserRequests: PropTypes.func.isRequired,
    storeUserRequests: PropTypes.func.isRequired,
    createRequest: PropTypes.func.isRequired,
    viewedRequest: PropTypes.func.isRequired,
    getRequestById: PropTypes.func.isRequired,
    updateRequest: PropTypes.func.isRequired,
    requests: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired,
    presentRequest: PropTypes.object.isRequired,
  };

  const mapStateToProps = state => ({
    requests: state.userRequests,
    presentRequest: state.currentRequest,
  });

  const mapDispatchToProps = dispatch => ({
    getUserRequests: () => dispatch(loadUserRequests()),
    storeUserRequests: requests => dispatch(userRequests(requests)),
    createRequest: request => dispatch(createUserRequest(request)),
    getRequestById: id => dispatch(getSingleRequest(id)),
    viewedRequest: request => dispatch(currentRequest(request)),
    updateRequest: (id, request) => dispatch(editUserRequest(id, request)),
  });

  return connect(mapStateToProps, mapDispatchToProps)(RequestComponent);
};

export default userPage;
