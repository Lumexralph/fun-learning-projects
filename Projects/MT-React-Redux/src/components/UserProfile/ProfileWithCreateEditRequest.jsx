import React from 'react';
import PropTypes from 'prop-types';

import requestHOC from '../HOC/withUserPage';

export const RequestForm = ({
  onChange, onSubmit, message, pathname, request,
}) => (
  <div id="request-modal" className="modal">
    <div className="modal-content">
      <div className="modal-header">
        <h2>FixZit</h2>
      </div>
      <div className="modal-body">
        <h3>{pathname === '/request/edit' ? 'Edit' : 'Create'} request</h3>
        <div>
          {message ? <p id="formPopup">{message}</p> : ''}
        </div>
        <form
          onSubmit={onSubmit}
        >

          <label htmlFor="title">
            <input
              onChange={e => onChange(e)}
              type="text"
              id="request-title"
              key={request !== null ? request.request_title : ''}
              name="title"
              placeholder="Your request title.."
              defaultValue={request !== null ? request.request_title : ''}
              required
            />
          </label>

          <label htmlFor="dept">
              Department
            <select
              name="department"
              onChange={e => onChange(e)}
              id="dept"
              key={request !== null ? request.department : ''}
              defaultValue={request !== null ? request.department : ''}
            >
              <option value="maintenance">Maintenance</option>
              <option value="Repairs">Repairs</option>
            </select>
          </label>


          <label htmlFor="content">
            <textarea
              onChange={e => onChange(e)}
              id="subject"
              name="content"
              placeholder="Tell us about your request.."
              style={{ height: '200px' }}
              key={request !== null ? request.request_content : ''}
              defaultValue={request !== null ? request.request_content : ''}
              required
            />
          </label>

          <button id="sendRequest" className="btn btn-send" type="submit">{pathname === '/request/edit' ? 'Update' : 'Send'} Request</button>

        </form>
      </div>
    </div>
  </div>
);

RequestForm.defaultProps = {
  request: null,
};

RequestForm.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  message: PropTypes.string.isRequired,
  pathname: PropTypes.string.isRequired,
  request: PropTypes.any,
};

const ProfileWithRequestForm = requestHOC(RequestForm);

export default ProfileWithRequestForm;
