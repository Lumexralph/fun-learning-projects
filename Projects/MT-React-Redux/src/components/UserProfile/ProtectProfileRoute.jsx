import React from 'react';
import { Redirect, Route } from 'react-router-dom';
import getDataFromStorage from '../../helpers/getDataFromStorage';

const ProtectProfileRoute = ({ component: Component, ...rest }) => {
  const admin = getDataFromStorage('admin');

  return (
    <Route
      {...rest}
      render={(props) => {
        if (admin === 'false') {
          return (
            <Component {...props} />
          );
        }
        return (
          <Redirect
            to={{
              pathname: '/login',
              state: { from: props.location },
            }}
          />
        );
      }
      }
    />);
};

export default ProtectProfileRoute;
