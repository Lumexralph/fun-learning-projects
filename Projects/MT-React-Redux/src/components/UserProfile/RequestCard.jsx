import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const RequestCard = ({ title, department, status, id }) => (
  <div className="flip-card">
    <div className="flip-card-inner">
      <div className="flip-card-front">
        <h1>{title}</h1>
      </div>
      <div className="flip-card-back">
        <h1>{title}</h1>
        <p><i className="fa fa-building" /> {department}</p>
        <p><i className="fa fa-flag-checkered" aria-hidden="true" />  {status}</p>
        <p><Link to={`/requests/${id}`}>Click to view</Link></p>
      </div>
    </div>
  </div>
);

RequestCard.propTypes = {
  title: PropTypes.string.isRequired,
  department: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default RequestCard;
