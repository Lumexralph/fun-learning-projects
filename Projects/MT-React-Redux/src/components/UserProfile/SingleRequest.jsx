import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import requestHOC from '../HOC/withUserPage';

export const SingleRequest = ({ request }) => (
  <div className="request-container">
    <h2>{request.request_title}</h2>
    <h4><i className="fa fa-file-text-o" aria-hidden="true" />  {request.request_content}</h4>
    <p><i className="fa fa-building" /> {request.department}</p>
    <p><i className="fa fa-flag-checkered" aria-hidden="true" /> {request.status}</p>
    <div>
      <button className="btn btn-nav" type="button">
        <Link to="/request/edit" style={{ textDecoration: 'none' }}><i className="fa fa-edit" /></Link>
      </button>
    </div>
  </div>
);

SingleRequest.propTypes = {
  request: PropTypes.object.isRequired,
};

const SingleRequestPage = requestHOC(SingleRequest);

export default SingleRequestPage;
