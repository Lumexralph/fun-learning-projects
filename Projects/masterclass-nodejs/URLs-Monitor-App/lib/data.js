/**
 * Lirbrary for storing and editing data
 *
 */

 // Dependencies
 const fs = require('fs');
 const path = require('path');
 const helpers = require('./helpers');

 // Container for the module to be exported
 const lib = { };

 // Base directory of the data folder
 // To make it a clean path
 lib.baseDirectory = path.join(__dirname, '../.data/');

// Write data to a file
lib.create = (dir, file, data, callback) => {
  // Open the files for writing
  // Using the path and mode for the file operation
  fs.open(`${lib.baseDirectory}${dir}/${file}.json`, 'wx', (err, fileDescriptor) => {
     if(!err && fileDescriptor) {
        // Convert data to string
      const stringData = JSON.stringify(data);

      // Write to the file and close it
      fs.writeFile(fileDescriptor, stringData, (err) => {
        if(!err) {
          fs.close(fileDescriptor, (err) => {
            if(!err) {
              callback(false);
            } else {
              callback('Error closing file');
            }
          });
        } else {
          callback('Error writing to the new file');
        }
      });
     } else {
      callback('Could not create new file, it could already exist');
     }
  });
};

// Read data from a a file
lib.read = (dir, file, callback) => {
  fs.readFile(`${lib.baseDirectory}${dir}/${file}.json`, 'utf8', (err, data) => {
    if(!err && data) {
      const parsedData = helpers.parseJsonToObject(data);
      callback(false, parsedData);
    } else {
      callback(err, data)
    }

  });
};

// Update an existing file
lib.update = (dir, file, data, callback) => {
  // Open the file for writing
  fs.open(`${lib.baseDirectory}${dir}/${file}.json`, 'r+', (err, fileDescriptor) => {
    if(!err && fileDescriptor) {
      const stringData = JSON.stringify(data);

      // To be able to update the file with existing data, the file has to be truncated
      fs.ftruncate(fileDescriptor, (err) => {
        if (!err) {
          // Write to the file and close it
          fs.writeFile(fileDescriptor, stringData, (err) => {
            if (!err) {
              // Close to the file
              fs.close(fileDescriptor, (err) => {
                if (!err) {
                  callback(false);
                } else {
                  callback('Error closing the existing file');
                }
              });
            } else {
              callback('Error writing data to existing file');
            }
          });
        } else {
          callback('Error occurred while truncating file');
        }
      });
    }
  });
}

// Delete a file
lib.delete = (dir, file, callback) => {
  // Unlink the file, remove it from the fi[e system
  fs.unlink(`${lib.baseDirectory}${dir}/${file}.json`, (err) => {
    if(!err) {
      callback('File successfully deleted.');
    } else {
      callback('Error deleting the file');
    }
  });
};

 // Export the module
 module.exports = lib;
