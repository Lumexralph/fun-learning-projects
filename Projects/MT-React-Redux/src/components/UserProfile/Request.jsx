import React from 'react';
import PropTypes from 'prop-types';

import RequestCard from './RequestCard';

const Request = ({ data }) => {
  if (data.length > 0) {
    const requestCards = data.map(request => (
      <div key={request.request_id}>
        <RequestCard
          title={request.request_title}
          department={request.department}
          status={request.status}
          id={request.request_id}
        />
      </div>
    ));

    return requestCards;
  }

  return (
    <div className="request-card">
      <p>No requests created yet, you can make a request!</p>
    </div>
  );
};

Request.propTypes = {
  data: PropTypes.array.isRequired,
};

export default Request;
