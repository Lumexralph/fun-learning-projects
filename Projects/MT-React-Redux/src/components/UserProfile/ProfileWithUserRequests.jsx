import React from 'react';
import PropTypes from 'prop-types';

import requestHOC from '../HOC/withUserPage';

import Request from './Request';

export const UserRequests = ({ requests }) => (
  <div className="card-container">
    <Request data={requests} />
  </div>
);

UserRequests.propTypes = {
  requests: PropTypes.array.isRequired,
};

export default requestHOC(UserRequests);
