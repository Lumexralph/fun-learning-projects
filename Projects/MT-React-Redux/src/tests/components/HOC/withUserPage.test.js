import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import withUserPage from '../../../components/HOC/withUserPage';
import { RequestForm } from '../../../components/UserProfile/ProfileWithCreateEditRequest';
import data from '../../__mocks__/requestsData';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const store = mockStore({
  userRequests: [{
    name: 'fixzit',
  }],
  currentRequest: {
    name: 'fixzit',
  },
});


// const renderer = new ShallowRenderer();

describe('Test for higher order component userpage', () => {
  const UserPage = withUserPage(RequestForm);

  const history = {
    location: {
      pathname: '/request/create',
    },
  };

  const request = {
    request_id: 47,
    request_title: 'Service the generator',
    request_content: 'Check if things are in order',
    department: 'maintenance',
    status: 'pending',
  };

  const match = {
    params: {
      id: '2',
    },
  };

  const getUserRequests = Promise.resolve(data);
  const getRequestById = () => Promise.resolve(request);
  const viewedRequest = e => e;
  const storeUserRequests = (resp = data) => resp;
  const createRequest = () => Promise.resolve(request);

  let component = renderer.create(
    <Router><UserPage
      store={store}
      history={history}
      match={match}
      requests={data}
      viewedRequest={viewedRequest}
      getRequestById={id => getRequestById(id)}
      getUserRequests={() => getUserRequests}
      storeUserRequests={storeUserRequests}
      createRequest={(requetsData => createRequest(requetsData))}
    />
    </Router>
    ,
  );

  let tree = component.toJSON();
  let componentInstance = component.root;

  it('Should return another component', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Should have an onSubmit meethod called in rendered component', (done) => {
    componentInstance.findByType('form').props.onSubmit({
      preventDefault: () => true,
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });

  it('Should call change event on username input field', (done) => {
    componentInstance.findByProps({ name: 'title' }).props.onChange({ target: { value: '' } });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });

  it('Should return another component with different data', (done) => {
    const history2 = {
      location: {
        pathname: '/',
      },
    };

    const match2 = {
      params: '',
    };
    component = renderer.create(
      <Router><UserPage
        store={store}
        history={history2}
        match={match2}
        requests={data}
        viewedRequest={viewedRequest}
        getRequestById={id => getRequestById(id)}
        getUserRequests={() => getUserRequests}
        storeUserRequests={storeUserRequests}
        createRequest={(requetsData => createRequest(requetsData))}
      />
      </Router>
      ,
    );
    tree = component.toJSON();
    componentInstance = component.root;
    expect(tree).toMatchSnapshot();
    done();
  });

  it('Should have an onSubmit meethod called in rendered component', (done) => {
    componentInstance.findByType('form').props.onSubmit({
      preventDefault: () => true,
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});
