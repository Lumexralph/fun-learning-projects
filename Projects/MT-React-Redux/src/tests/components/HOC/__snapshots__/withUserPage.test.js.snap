// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for higher order component userpage Should call change event on username input field 1`] = `
<div
  className="container"
>
  <header>
    <img
      alt="header logo"
      className="header-logo"
      src="test-file-stub"
    />
    <nav>
      <ul
        className="nav"
      >
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/profile"
            onClick={[Function]}
          >
            My Requests
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/request/create"
            onClick={[Function]}
          >
            Make A Request
          </a>
        </li>
        <li>
          <button
            className="btn nav-link nav-btn"
            onClick={[Function]}
            style={
              Object {
                "background": "inherit",
              }
            }
            type="button"
          >
            Logout
          </button>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="modal"
    id="request-modal"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h2>
          FixZit
        </h2>
      </div>
      <div
        className="modal-body"
      >
        <h3>
          Create
           request
        </h3>
        <div>
          
        </div>
        <form
          onSubmit={[Function]}
        >
          <label
            htmlFor="title"
          >
            <input
              defaultValue=""
              id="request-title"
              name="title"
              onChange={[Function]}
              placeholder="Your request title.."
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="dept"
          >
            Department
            <select
              defaultValue=""
              id="dept"
              name="department"
              onChange={[Function]}
            >
              <option
                value="maintenance"
              >
                Maintenance
              </option>
              <option
                value="Repairs"
              >
                Repairs
              </option>
            </select>
          </label>
          <label
            htmlFor="content"
          >
            <textarea
              defaultValue=""
              id="subject"
              name="content"
              onChange={[Function]}
              placeholder="Tell us about your request.."
              required={true}
              style={
                Object {
                  "height": "200px",
                }
              }
            />
          </label>
          <button
            className="btn btn-send"
            id="sendRequest"
            type="submit"
          >
            Send
             Request
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test for higher order component userpage Should have an onSubmit meethod called in rendered component 1`] = `
<div
  className="container"
>
  <header>
    <img
      alt="header logo"
      className="header-logo"
      src="test-file-stub"
    />
    <nav>
      <ul
        className="nav"
      >
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/profile"
            onClick={[Function]}
          >
            My Requests
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/request/create"
            onClick={[Function]}
          >
            Make A Request
          </a>
        </li>
        <li>
          <button
            className="btn nav-link nav-btn"
            onClick={[Function]}
            style={
              Object {
                "background": "inherit",
              }
            }
            type="button"
          >
            Logout
          </button>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="modal"
    id="request-modal"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h2>
          FixZit
        </h2>
      </div>
      <div
        className="modal-body"
      >
        <h3>
          Create
           request
        </h3>
        <div>
          
        </div>
        <form
          onSubmit={[Function]}
        >
          <label
            htmlFor="title"
          >
            <input
              defaultValue=""
              id="request-title"
              name="title"
              onChange={[Function]}
              placeholder="Your request title.."
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="dept"
          >
            Department
            <select
              defaultValue=""
              id="dept"
              name="department"
              onChange={[Function]}
            >
              <option
                value="maintenance"
              >
                Maintenance
              </option>
              <option
                value="Repairs"
              >
                Repairs
              </option>
            </select>
          </label>
          <label
            htmlFor="content"
          >
            <textarea
              defaultValue=""
              id="subject"
              name="content"
              onChange={[Function]}
              placeholder="Tell us about your request.."
              required={true}
              style={
                Object {
                  "height": "200px",
                }
              }
            />
          </label>
          <button
            className="btn btn-send"
            id="sendRequest"
            type="submit"
          >
            Send
             Request
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test for higher order component userpage Should have an onSubmit meethod called in rendered component 2`] = `
<div
  className="container"
>
  <header>
    <img
      alt="header logo"
      className="header-logo"
      src="test-file-stub"
    />
    <nav>
      <ul
        className="nav"
      >
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/profile"
            onClick={[Function]}
          >
            My Requests
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/request/create"
            onClick={[Function]}
          >
            Make A Request
          </a>
        </li>
        <li>
          <button
            className="btn nav-link nav-btn"
            onClick={[Function]}
            style={
              Object {
                "background": "inherit",
              }
            }
            type="button"
          >
            Logout
          </button>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="modal"
    id="request-modal"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h2>
          FixZit
        </h2>
      </div>
      <div
        className="modal-body"
      >
        <h3>
          Create
           request
        </h3>
        <div>
          
        </div>
        <form
          onSubmit={[Function]}
        >
          <label
            htmlFor="title"
          >
            <input
              id="request-title"
              name="title"
              onChange={[Function]}
              placeholder="Your request title.."
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="dept"
          >
            Department
            <select
              id="dept"
              name="department"
              onChange={[Function]}
            >
              <option
                value="maintenance"
              >
                Maintenance
              </option>
              <option
                value="Repairs"
              >
                Repairs
              </option>
            </select>
          </label>
          <label
            htmlFor="content"
          >
            <textarea
              id="subject"
              name="content"
              onChange={[Function]}
              placeholder="Tell us about your request.."
              required={true}
              style={
                Object {
                  "height": "200px",
                }
              }
            />
          </label>
          <button
            className="btn btn-send"
            id="sendRequest"
            type="submit"
          >
            Send
             Request
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test for higher order component userpage Should return another component 1`] = `
<div
  className="container"
>
  <header>
    <img
      alt="header logo"
      className="header-logo"
      src="test-file-stub"
    />
    <nav>
      <ul
        className="nav"
      >
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/profile"
            onClick={[Function]}
          >
            My Requests
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/request/create"
            onClick={[Function]}
          >
            Make A Request
          </a>
        </li>
        <li>
          <button
            className="btn nav-link nav-btn"
            onClick={[Function]}
            style={
              Object {
                "background": "inherit",
              }
            }
            type="button"
          >
            Logout
          </button>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="modal"
    id="request-modal"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h2>
          FixZit
        </h2>
      </div>
      <div
        className="modal-body"
      >
        <h3>
          Create
           request
        </h3>
        <div>
          
        </div>
        <form
          onSubmit={[Function]}
        >
          <label
            htmlFor="title"
          >
            <input
              defaultValue=""
              id="request-title"
              name="title"
              onChange={[Function]}
              placeholder="Your request title.."
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="dept"
          >
            Department
            <select
              defaultValue=""
              id="dept"
              name="department"
              onChange={[Function]}
            >
              <option
                value="maintenance"
              >
                Maintenance
              </option>
              <option
                value="Repairs"
              >
                Repairs
              </option>
            </select>
          </label>
          <label
            htmlFor="content"
          >
            <textarea
              defaultValue=""
              id="subject"
              name="content"
              onChange={[Function]}
              placeholder="Tell us about your request.."
              required={true}
              style={
                Object {
                  "height": "200px",
                }
              }
            />
          </label>
          <button
            className="btn btn-send"
            id="sendRequest"
            type="submit"
          >
            Send
             Request
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test for higher order component userpage Should return another component with different data 1`] = `
<div
  className="container"
>
  <header>
    <img
      alt="header logo"
      className="header-logo"
      src="test-file-stub"
    />
    <nav>
      <ul
        className="nav"
      >
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/profile"
            onClick={[Function]}
          >
            My Requests
          </a>
        </li>
        <li>
          <a
            className="btn nav-btn nav-link"
            href="/request/create"
            onClick={[Function]}
          >
            Make A Request
          </a>
        </li>
        <li>
          <button
            className="btn nav-link nav-btn"
            onClick={[Function]}
            style={
              Object {
                "background": "inherit",
              }
            }
            type="button"
          >
            Logout
          </button>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="modal"
    id="request-modal"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h2>
          FixZit
        </h2>
      </div>
      <div
        className="modal-body"
      >
        <h3>
          Create
           request
        </h3>
        <div>
          
        </div>
        <form
          onSubmit={[Function]}
        >
          <label
            htmlFor="title"
          >
            <input
              id="request-title"
              name="title"
              onChange={[Function]}
              placeholder="Your request title.."
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="dept"
          >
            Department
            <select
              id="dept"
              name="department"
              onChange={[Function]}
            >
              <option
                value="maintenance"
              >
                Maintenance
              </option>
              <option
                value="Repairs"
              >
                Repairs
              </option>
            </select>
          </label>
          <label
            htmlFor="content"
          >
            <textarea
              id="subject"
              name="content"
              onChange={[Function]}
              placeholder="Tell us about your request.."
              required={true}
              style={
                Object {
                  "height": "200px",
                }
              }
            />
          </label>
          <button
            className="btn btn-send"
            id="sendRequest"
            type="submit"
          >
            Send
             Request
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
